File: ployglots/mta-ployglot-pkg-vault/packages/ts/src/sdk.ts
// @ts-nocheck
import type { IVaultFileLogger } from './logger.js';
import { Logger } from './logger.js';
import type {
  SDKResult,
  ConfigDescription,
  SecretInfo,
  ValidationResult,
  DiagnosticResult,
  IVaultFileSDK,
} from './sdk-types.js';
import { makeSDKError } from './sdk-types.js';
import type { LoadResult } from './domain.js';
import { EnvStore } from './env-store.js';
import { parseEnvFile } from './core.js';

export class VaultFileSDK implements IVaultFileSDK {
  private envPath: string | undefined;
  private base64Parsers: Record<string, (s: string) => string> = {};
  private logger: IVaultFileLogger;

  private constructor(logger?: IVaultFileLogger) {
    this.logger = logger ?? Logger.create('vault-file', 'sdk');
  }

  static create(): VaultFileSDKBuilder {
    return new VaultFileSDKBuilder();
  }

  /** Internal factory used by VaultFileSDKBuilder.build(). Not public. */
  static newForBuilder(): VaultFileSDK {
    return new VaultFileSDK();
  }

  setEnvPath(path: string): void {
    this.envPath = path;
  }
  setBase64Parsers(parsers: Record<string, (s: string) => string>): void {
    this.base64Parsers = parsers;
  }
  setLogger(logger: IVaultFileLogger): void {
    this.logger = logger;
  }

  protected success<T>(data: T): SDKResult<T> {
    return { success: true, data };
  }
  protected failure<T>(
    code: string,
    message: string,
    details?: Record<string, unknown>,
  ): SDKResult<T> {
    return { success: false, error: makeSDKError(code, message, details) };
  }

  loadConfig(): SDKResult<LoadResult> {
    try {
      const result = EnvStore.onStartup(this.envPath ?? '.env', this.logger);
      return this.success(result);
    } catch (err) {
      return this.failure('LOAD_FAILED', (err as Error).message, { envPath: this.envPath });
    }
  }

  loadFromPath(filePath: string): SDKResult<LoadResult> {
    try {
      const result = EnvStore.onStartup(filePath, this.logger);
      return this.success(result);
    } catch (err) {
      return this.failure('LOAD_FAILED', (err as Error).message, { filePath });
    }
  }

  validateFile(filePath: string): SDKResult<ValidationResult> {
    const errors: string[] = [];
    const warnings: string[] = [];
    try {
      const parsed = parseEnvFile(filePath);
      if (Object.keys(parsed).length === 0) warnings.push('file parsed to empty map');
      return this.success({ valid: errors.length === 0, errors, warnings });
    } catch (err) {
      errors.push((err as Error).message);
      return this.success({ valid: false, errors, warnings });
    }
  }

  /**
   * @stub Not yet implemented. Returns SDKResult with error.code 'NOT_IMPLEMENTED'.
   * Declared on IVaultFileSDK so consumers can program against the final shape.
   */
  exportToFormat(_format: 'json' | 'yaml', _path: string): SDKResult<void> {
    return this.failure('NOT_IMPLEMENTED', 'exportToFormat is a stub');
  }

  describeConfig(): SDKResult<ConfigDescription> {
    const varsCount = EnvStore._getTotalVarsLoaded();
    return this.success({
      version: '1.0.0',
      varsCount,
      source: this.envPath ?? '.env',
    });
  }

  getSecretSafe(key: string): SDKResult<SecretInfo> {
    const value = EnvStore.get(key);
    if (value === undefined) {
      return this.failure('KEY_NOT_FOUND', `key '${key}' not present`);
    }
    return this.success({ key, masked: '***', exists: true });
  }

  /** @stub Returns an empty list. */
  listAvailableKeys(): SDKResult<string[]> {
    return this.success([]);
  }

  diagnoseEnvStore(): SDKResult<DiagnosticResult> {
    return this.success({
      initialized: EnvStore.isInitialized(),
      varsLoaded: EnvStore._getTotalVarsLoaded(),
      source: this.envPath,
    });
  }

  findMissingRequired(keys: string[]): SDKResult<string[]> {
    const missing = keys.filter((k) => EnvStore.get(k) === undefined);
    return this.success(missing);
  }

  /** @stub Returns an empty list. */
  suggestMissingKeys(_partial: string): SDKResult<string[]> {
    return this.success([]);
  }
}

export class VaultFileSDKBuilder {
  private envPath: string | undefined;
  private base64Parsers: Record<string, (s: string) => string> = {};
  private logger: IVaultFileLogger | undefined;

  withEnvPath(path: string): VaultFileSDKBuilder {
    this.envPath = path;
    return this;
  }

  withBase64Parsers(parsers: Record<string, (s: string) => string>): VaultFileSDKBuilder {
    this.base64Parsers = parsers;
    return this;
  }

  withLogger(logger: IVaultFileLogger): VaultFileSDKBuilder {
    this.logger = logger;
    return this;
  }

  build(): VaultFileSDK {
    const sdk = VaultFileSDK.newForBuilder();
    if (this.envPath !== undefined) sdk.setEnvPath(this.envPath);
    sdk.setBase64Parsers(this.base64Parsers);
    sdk.setLogger(this.logger ?? Logger.create('vault-file', 'sdk'));
    return sdk;
  }
}

File: ployglots/mta-ployglot-pkg-vault/packages/py/src/polyglot_vault_file/sdk.py
"""VaultFileSDK + VaultFileSDKBuilder."""
from __future__ import annotations

from typing import Any, Callable, Dict, List, Literal, Optional

from .core import parse_env_file
from .env_store import EnvStore
from .logger import IVaultFileLogger, Logger
from .sdk_types import (
    ConfigDescription,
    DiagnosticResult,
    SDKError,
    SDKResult,
    SecretInfo,
    ValidationResult,
)


class VaultFileSDK:
    def __init__(self, logger: Optional[IVaultFileLogger] = None) -> None:
        self._env_path: Optional[str] = None
        self._base64_parsers: Dict[str, Callable[[str], str]] = {}
        self._logger: IVaultFileLogger = logger or Logger.create("polyglot_vault_file", "sdk")

    @classmethod
    def create(cls) -> "VaultFileSDKBuilder":
        return VaultFileSDKBuilder()

    def set_env_path(self, path: str) -> None:
        self._env_path = path

    def set_base64_parsers(self, parsers: Dict[str, Callable[[str], str]]) -> None:
        self._base64_parsers = parsers

    def set_logger(self, logger: IVaultFileLogger) -> None:
        self._logger = logger

    def _success(self, data: Any) -> SDKResult:
        return SDKResult(success=True, data=data)

    def _failure(self, code: str, message: str, details: Optional[Dict[str, Any]] = None) -> SDKResult:
        return SDKResult(success=False, error=SDKError(code=code, message=message, details=details))

    def load_config(self) -> SDKResult:
        try:
            result = EnvStore.on_startup(self._env_path or ".env", self._logger)
            return self._success(result)
        except Exception as err:
            return self._failure("LOAD_FAILED", str(err), {"envPath": self._env_path})

    def load_from_path(self, file_path: str) -> SDKResult:
        try:
            result = EnvStore.on_startup(file_path, self._logger)
            return self._success(result)
        except Exception as err:
            return self._failure("LOAD_FAILED", str(err), {"filePath": file_path})

    def validate_file(self, file_path: str) -> SDKResult:
        errors: List[str] = []
        warnings: List[str] = []
        try:
            parsed = parse_env_file(file_path)
            if not parsed:
                warnings.append("file parsed to empty map")
            return self._success(
                ValidationResult(valid=len(errors) == 0, errors=errors, warnings=warnings)
            )
        except Exception as err:
            errors.append(str(err))
            return self._success(ValidationResult(valid=False, errors=errors, warnings=warnings))

    def export_to_format(self, format: Literal["json", "yaml"], path: str) -> SDKResult:
        """Stub — returns SDKResult with error.code 'NOT_IMPLEMENTED'."""
        return self._failure("NOT_IMPLEMENTED", "export_to_format is a stub")

    def describe_config(self) -> SDKResult:
        vars_count = EnvStore._get_total_vars_loaded()
        return self._success(
            ConfigDescription(
                version="1.0.0",
                vars_count=vars_count,
                source=self._env_path or ".env",
            )
        )

    def get_secret_safe(self, key: str) -> SDKResult:
        value = EnvStore.get(key)
        if value is None:
            return self._failure("KEY_NOT_FOUND", f"key '{key}' not present")
        return self._success(SecretInfo(key=key, masked="***", exists=True))

    def list_available_keys(self) -> SDKResult:
        """Stub — returns empty list."""
        return self._success([])

    def diagnose_env_store(self) -> SDKResult:
        return self._success(
            DiagnosticResult(
                initialized=EnvStore.is_initialized(),
                vars_loaded=EnvStore._get_total_vars_loaded(),
                source=self._env_path,
            )
        )

    def find_missing_required(self, keys: List[str]) -> SDKResult:
        missing = [k for k in keys if EnvStore.get(k) is None]
        return self._success(missing)

    def suggest_missing_keys(self, partial: str) -> SDKResult:
        """Stub — returns empty list."""
        return self._success([])


class VaultFileSDKBuilder:
    def __init__(self) -> None:
        self._env_path: Optional[str] = None
        self._base64_parsers: Dict[str, Callable[[str], str]] = {}
        self._logger: Optional[IVaultFileLogger] = None

    def with_env_path(self, path: str) -> "VaultFileSDKBuilder":
        self._env_path = path
        return self

    def with_base64_parsers(self, parsers: Dict[str, Callable[[str], str]]) -> "VaultFileSDKBuilder":
        self._base64_parsers = parsers
        return self

    def with_logger(self, logger: IVaultFileLogger) -> "VaultFileSDKBuilder":
        self._logger = logger
        return self

    def build(self) -> VaultFileSDK:
        sdk_logger = (
            self._logger
            if self._logger is not None
            else Logger.create("polyglot_vault_file", "sdk")
        )
        sdk = VaultFileSDK(logger=sdk_logger)
        if self._env_path is not None:
            sdk.set_env_path(self._env_path)
        sdk.set_base64_parsers(self._base64_parsers)
        return sdk
